docs/basics.adoc

```diff
@@ -329,6 +329,38 @@ Note, that schema name changes may not be immediately visible in currently runni
 Should the need arise to re-define an existing schema type, it is recommended to change the name of this type to a name that is not currently (and will never be) in use. After that, a new label or key can be defined with the original name, thereby effectively replacing the old one.
 However, note that this would not affect vertices, edges, or properties previously written with the existing type. Redefining existing graph elements is not supported online and must be accomplished through a batch graph transformation.
 
+=== Schema constraints
+
+The definition of the schema allows to configure explict property and connection constraints. However, these constraints are disabled by default. These constraints can be forced to applyied by setting `schema.constraints=true` in the JanusGraph graph configuration.
```

Comment:
Add some discussion on why a developer might want to add constraints. Also need to mention the `schema.default` behavior here.